### PR TITLE
Fix creating a target right after AppCenter.start

### PR DESCRIPTION
**Pull request by guperrot**
Use the fact that any operation are queued into background looper so it is guaranteed that the `initInBackground` will be called in a sequence after the `Analytics.onStarted` that set context/channel to non null, in other terms if you execute anything in post, it will always have non null channel/context.

### CODE REVIEW

File: sdk/appcenter-analytics/src/main/java/com/microsoft/appcenter/analytics/PropertyConfigurator.java [hunk 1]
```diff
@@ -208,9 +212,12 @@ public synchronized void removeEventProperty(String key) {
...
     }
 
-    @SuppressLint("HardwareIds")
+    /**
+     * Enable collection of the Android device identifier for this target.
+     * This does not have any effect on child transmission targets.
+     */
     public void collectDeviceId() {
-        mDeviceId = Secure.getString(mTransmissionTarget.mContext.getContentResolver(), Secure.ANDROID_ID);
+        mDeviceIdEnabled = true;
```
Comment (guperrot) [inline sdk/appcenter-analytics/src/main/java/com/microsoft/appcenter/analytics/PropertyConfigurator.java]: Context can be null when calling this early so we just remember the call and apply later.

File: sdk/appcenter-analytics/src/test/java/com/microsoft/appcenter/analytics/AnalyticsTest.java [hunk 2]
```diff
@@ -507,12 +507,17 @@ public boolean matches(Object item) {
...
+        when(AppCenter.isConfigured()).thenReturn(false);
+
+        /* When creating a target, it returns null. */
         assertNull(Analytics.getTransmissionTarget("t1"));
 
-        /* Verify log. */
+        /* And prints an error. */
         verifyStatic();
-        AppCenterLog.error(anyString(), contains("App context is null, App Center has not been started."));
+        AppCenterLog.error(anyString(), contains("AppCenter is not configured"));
```
Comment (jwallra) [inline sdk/appcenter-analytics/src/test/java/com/microsoft/appcenter/analytics/AnalyticsTest.java]: Add period at end of log message

File: apps/sasquatch/src/main/java/com/microsoft/appcenter/sasquatch/activities/EventActivity.java [hunk 3]
```diff
@@ -48,6 +48,9 @@
     public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
 
+        /* Test start from library. */
+        AppCenter.startFromLibrary(this, Analytics.class);
+
```
Comment (guperrot) [inline apps/sasquatch/src/main/java/com/microsoft/appcenter/sasquatch/activities/EventActivity.java]: SDK checks SDK started before you can create targets so had to move earlier.

File: sdk/appcenter-analytics/src/main/java/com/microsoft/appcenter/analytics/Analytics.java [hunk 4]
```diff
@@ -624,7 +633,9 @@ public void onConfigurationUpdated(String appSecret, String transmissionTargetTo
      */
     @WorkerThread
     private void setDefaultTransmissionTarget(String transmissionTargetToken) {
-        mDefaultTransmissionTarget = getInstanceTransmissionTarget(transmissionTargetToken);
+        if (transmissionTargetToken != null) {
+            mDefaultTransmissionTarget = getInstanceTransmissionTarget(transmissionTargetToken);
```
Comment (guperrot) [inline sdk/appcenter-analytics/src/main/java/com/microsoft/appcenter/analytics/Analytics.java]: This is already covered in unit test. Without the if, mDefaultTransmissionTarget would still be null but with an error log saying that you need to pass non null. Since we do it internally its confusing for users (this happens when starting with no secret from an app).

File: sdk/appcenter-analytics/src/test/java/com/microsoft/appcenter/analytics/AnalyticsTest.java (hunk 2 above)
Comment (guperrot) [reply sdk/appcenter-analytics/src/test/java/com/microsoft/appcenter/analytics/AnalyticsTest.java]: its not the log but the test verification with contains a substring

Comment (jwallra) [reply sdk/appcenter-analytics/src/test/java/com/microsoft/appcenter/analytics/AnalyticsTest.java]: Whoops, should pay closer attention to what I'm looking at 😅

File: sdk/appcenter-analytics/src/main/java/com/microsoft/appcenter/analytics/PropertyConfigurator.java (hunk 1 above)
Comment (jaelim-ms) [reply sdk/appcenter-analytics/src/main/java/com/microsoft/appcenter/analytics/PropertyConfigurator.java]: Understand this change but we aligned the iOS implementation matching with Android.
Not a big deal but we need to change it in iOS, too.

File: sdk/appcenter-analytics/src/main/java/com/microsoft/appcenter/analytics/Analytics.java [hunk 5]
```diff
@@ -59,7 +61,8 @@
     /**
      * Shared instance.
      */
-    private static Analytics sInstance = null;
+    @SuppressLint("StaticFieldLeak")
+    private static Analytics sInstance;
```
Comment (clpolet) [inline sdk/appcenter-analytics/src/main/java/com/microsoft/appcenter/analytics/Analytics.java]: Just curious, what's the reason for this change?

**Comment by jaelim-ms** [comment]
Isn't is a fix for the comment that you mentioned in #797?

File: sdk/appcenter-analytics/src/main/java/com/microsoft/appcenter/analytics/Analytics.java (hunk 5 above)
Comment (guperrot) [reply sdk/appcenter-analytics/src/main/java/com/microsoft/appcenter/analytics/Analytics.java]: It was a warning we always ignore, also no need to init with null as its default value.

Comment (jaelim-ms) [reply sdk/appcenter-analytics/src/main/java/com/microsoft/appcenter/analytics/Analytics.java]: We don't initialize anything to null since it is by default. And it added `SuppressLint` to ignore lint complains.

**Comment by guperrot** [comment]
@jaelim-ms yes I should close #797 after merging this.

File: sdk/appcenter-analytics/src/main/java/com/microsoft/appcenter/analytics/PropertyConfigurator.java (hunk 1 above)
Comment (guperrot) [reply sdk/appcenter-analytics/src/main/java/com/microsoft/appcenter/analytics/PropertyConfigurator.java]: I don't think we should change iOS as this constraint is a bug fix for Android.
